raiden/network/rpc/smartcontract_proxy.py

```diff
@@ -2,75 +2,47 @@
 from ethereum.abi import ContractTranslator
 from ethereum.utils import normalize_address
 
+from raiden.exceptions import InvalidFunctionName
 
-class ContractProxy(object):
-    """ Exposes a smart contract as a python object.
-
-    Contract calls can be made directly in this object, all the functions will
-    be exposed with the equivalent api and will perform the argument
-    translation.
-    """
-
-    def __init__(self, sender, abi, address, call_func, transact_func, estimate_function=None):
-        sender = normalize_address(sender)
-
-        self.abi = abi
-        self.address = address = normalize_address(address)
-        self.translator = ContractTranslator(abi)
-
-        for function_name in self.translator.function_data:
-            function_proxy = MethodProxy(
-                sender,
-                address,
-                function_name,
-                self.translator,
-                call_func,
-                transact_func,
-                estimate_function,
-            )
 
-            type_argument = self.translator.function_data[function_name]['signature']
+VALID_KARGS = {'gasprice', 'startgas', 'value'}
```

Comment:
Is this used somewhere else? Otherwise it should be a static member

Reply:
I there any advantage of the global over the old approach?: https://github.com/hackaugusto/raiden/blob/ca2564bb/raiden/network/rpc/smartcontract_proxy.py#L50

Reply:
Not that I can think of.

Reply:
Let's switch this to the class member approach then. Then this PR is good to go.